### PR TITLE
chore: add loadgen

### PR DESCRIPTION
**Pull request by pgautier404**
This commit adds a load generator and benchmarking program. Sample summary output from a complete run with default settings:

```
=====> run complete <====
==============================
cumulative stats:
      total requests: 6000 (99 tps)
             success: 6000 (100%) (99 tps)
         unavailable: 0 (0%)
    timeout exceeded: 0 (0%)
      limit exceeded: 0 (0%)

cumulative get latencies:
      total requests: 3000
                 p50: 31
                 p90: 37
                 p99: 43
               p99.9: 45
                 max: 51

cumulative set latencies:
      total requests: 3000
                 p50: 31
                 p90: 39
                 p99: 53
               p99.9: 67
                 max: 67

completed in 60.104346 seconds
```

### CODE REVIEW

File: examples/loadgen/main.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+			startTime = hrtime.Now()
+		}
+
+		select {
+		case <-ctx.Done():
+			fmt.Println("\n=====> run complete <====")
+			printStats(getHistogram, setHistogram, errorCounter, origStartTime)
+			return
+		case getMessage := <-getChan:
+			if err := getHistogram.RecordValue(getMessage); err != nil {
```
Comment (cprice404) [inline examples/loadgen/main.go:254]: oh nice, so all the messages that cause us to update the histo come in on a channel and are handled by this one worker, correct?  So we don't have to worry about the histo being threadsafe?

**Review by cprice404 — COMMENTED** [review]
looks good!  one question about the histo.  Also can you share some of the numbers or output you are seeing when you run it?

File: examples/loadgen/main.go (hunk 1 above)
Comment (pgautier404) [reply examples/loadgen/main.go:254]: Yep, exactly.

**Review by cprice404 — APPROVED** [review]
looks like needs a rebase, and would love to see sample output, but I'm 👍 now

**Comment by cprice404** [comment]
@pgautier404 can you copy/paste some console output into the PR description?

**Comment by cprice404** [comment]
thanks.  those latencies look a little on the high side but maybe fine for laptop.  we should probably do a quick test running this loadgen from an ec2 box in-region next week.

**Review by cprice404 — APPROVED** [review]

**Pull request merged**